samcli/lib/build/build_graph.py

```diff
@@ -115,54 +187,122 @@ def _read(self):
         Each build definition will have empty function list, which will be populated from the current template.yaml file
         """
         LOG.debug("Instantiating build definitions")
-        self._build_definitions = []
+        self._function_build_definitions = []
+        self._layer_build_definitions = []
         document = {}
         try:
             txt = self._filepath.read_text()
             document = tomlkit.loads(txt)
         except OSError:
             LOG.debug("No previous build graph found, generating new one")
-        build_definitions_table = document.get(BuildGraph.BUILD_DEFINITIONS, [])
-        for build_definition_key in build_definitions_table:
-            build_definition = _toml_table_to_build_definition(build_definition_key,
-                                                               build_definitions_table[build_definition_key])
-            self._build_definitions.append(build_definition)
-
-        return self._build_definitions
+        function_build_definitions_table = document.get(BuildGraph.FUNCTION_BUILD_DEFINITIONS, [])
+        for function_build_definition_key in function_build_definitions_table:
+            function_build_definition = _toml_table_to_function_build_definition(function_build_definition_key,
+                                                               function_build_definitions_table[
+                                                                   function_build_definition_key])
+            self._function_build_definitions.append(function_build_definition)
+
+        layer_build_definitions_table = document.get(BuildGraph.LAYER_BUILD_DEFINITIONS, [])
+        for layer_build_definition_key in layer_build_definitions_table:
+            layer_build_definition = _toml_table_to_layer_build_definition(layer_build_definition_key,
+                                                                     layer_build_definitions_table[
+                                                                         layer_build_definition_key])
+            self._layer_build_definitions.append(layer_build_definition)
+
+        # return self._build_definitions
```

Comment:
we can remove statement if it is not needed anymore

Reply:
removed it.